matrix_factorization/training.h

```diff
@@ -8,6 +8,8 @@
 
 using namespace cu2rec;
 
+void train(CudaCSRMatrix* matrix, config::Config* cfg, float **P_ptr, float **Q_ptr, float *Q, float **losses_ptr,
```

Comment:
can we add some comments to these functions to say which is which (train vs partial fit). Hard to guess from just looking at the function sigs